driver-core/src/main/com/mongodb/connection/ReplyMessage.java

```diff
@@ -43,8 +43,8 @@
         this(responseBuffers.getReplyHeader(), requestId);
 
         if (replyHeader.getNumberReturned() > 0) {
+            BsonInput bsonInput = new ByteBufferBsonInput(responseBuffers.getBodyByteBuffer().duplicate());
```

Comment:
Now that the BsonInput is closed in the finally block, the ByteBuf it wraps has to be duplicated to avoid the original being released.

Reply:
Ack